skyportal/facility_apis/observation_plan.py

```diff
@@ -144,15 +150,52 @@ def generate_plan(observation_plan_id, request_id):
             params, is3D=params["do3D"], map_struct=params['map_struct']
         )
 
-        moc_structs = gwemopt.skyportal.create_moc_from_skyportal(
-            params, map_struct=map_struct
-        )
-        tile_structs = gwemopt.tiles.moc(params, map_struct, moc_structs)
+        if params["tilesType"] == "galaxy":
+            query = Galaxy.query_records_accessible_by(user, mode="read")
+            query = query.filter(Galaxy.catalog_name == params["galaxy_catalog"])
+            galaxies = query.all()
+            catalog_struct = {}
+            catalog_struct["ra"] = np.array([g.ra for g in galaxies])
+            catalog_struct["dec"] = np.array([g.dec for g in galaxies])
+            catalog_struct["S"] = np.array([1.0 for g in galaxies])
+            catalog_struct["Sloc"] = np.array([1.0 for g in galaxies])
+            catalog_struct["Smass"] = np.array([1.0 for g in galaxies])
```

Comment:
Did you intend to leave 1.0 as the value for all these?

Reply:
Yeah because will need to think about how to properly treat galaxy metric weighting.